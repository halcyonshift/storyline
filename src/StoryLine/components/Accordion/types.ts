import { ReactElement } from 'react'

export type AccordionProps = {
    id: string
    title: ReactElement
    children: ReactElement
    className?: string
    sx?: object
}
