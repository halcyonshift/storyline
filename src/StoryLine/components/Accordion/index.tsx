import { useState } from 'react'
import ArrowForwardIosSharpIcon from '@mui/icons-material/ArrowForwardIosSharp'
import Box from '@mui/material/Box'

import { AccordionProps } from './types'

const Accordion = ({ id, title, children, className, ...props }: AccordionProps) => {
    const [open, setOpen] = useState<boolean>(false)

    return (
        <>
            <Box
                className={`flex items-center ${className}`}
                {...props}
                role='button'
                aria-expanded={open}
                aria-controls={`content-${id}`}
                onClick={() => setOpen(!open)}>
                <ArrowForwardIosSharpIcon
                    className='mr-1'
                    sx={{ fontSize: '0.8rem', transform: open ? 'rotate(90deg)' : '' }}
                />
                {title}
            </Box>
            {open ? <Box id={`content-${id}`}>{children}</Box> : null}
        </>
    )
}

export default Accordion
