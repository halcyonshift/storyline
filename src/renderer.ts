import { init } from '@sentry/electron/renderer'
import { init as reactInit } from '@sentry/react'

if (parseInt(process.env.DEBUG) === 0) {
    init({ dsn: process.env.SENTRY }, reactInit)
}

import '@fontsource/roboto/300.css'
import '@fontsource/roboto/400.css'
import '@fontsource/roboto/500.css'
import '@fontsource/roboto/700.css'
import './index.css'
import './StoryLine'
