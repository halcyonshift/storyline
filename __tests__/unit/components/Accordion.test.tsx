import { Typography } from '@mui/material'
import userEvent from '@testing-library/user-event'
import renderer from 'react-test-renderer'
import Accordion from '@sl/components/Accordion'
import { render, screen } from '../test-utils'

describe('<Accordion />', () => {
    const accordion = (
        <Accordion id='1' title={<Typography>Title</Typography>}>
            <Typography>Show me</Typography>
        </Accordion>
    )
    beforeEach(async () => {
        render(accordion)
    })

    it('is closed by default', async () => {
        expect(screen.getByText('Title')).toBeTruthy()
        expect(screen.queryByText('Show me')).toBeFalsy()
    })

    it('opens on title click', async () => {
        await userEvent.click(screen.getByRole('button'))
        expect(screen.getByText('Show me')).toBeTruthy()
    })

    it('renders correctly', () => {
        const tree = renderer.create(accordion).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
