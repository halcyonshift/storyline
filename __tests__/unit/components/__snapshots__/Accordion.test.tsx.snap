// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders correctly 1`] = `
<div
  aria-controls="content-1"
  aria-expanded={false}
  className="flex items-center undefined MuiBox-root css-0"
  onClick={[Function]}
  role="button"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mr-1 css-1mq2mph-MuiSvgIcon-root"
    data-testid="ArrowForwardIosSharpIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
    />
  </svg>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Title
  </p>
</div>
`;
